### PR TITLE
Default remote setting

### PR DESCRIPTION
**Pull request by matthewnitschke**
# Problem
When multiple remotes are returned from `git remote`, this extension defaults to using the first returned remote name. This can be frustrating when the first remote name is not the desired remote to use.

# Changes
Adds a `sourcegraph.defaultRemote` setting to enable specifying a static remote that can be use. When `sourcegraph.defaultRemote` is null (the default setting value), the old remote logic is used.

### CODE REVIEW

File: .vscode/tasks.json [hunk 1]
```diff
@@ -9,5 +9,10 @@
         "reveal": "never",
       },
     },
+    {
+      "type": "npm",
+      "label": "build",
+      "script": "build"
+    }
```
Comment (matthewnitschke) [inline .vscode/tasks.json:16]: Running the `Launch Setting` task resulted in an error. Adding this task fixed the issue

File: .vscode/tasks.json [hunk 2]
```diff
@@ -9,5 +9,10 @@
         "reveal": "never",
       },
     },
+    {
+      "type": "npm",
+      "label": "build",
+      "script": "build",
+    },
```
Comment (matthewnitschke) [inline .vscode/tasks.json:16]: Running the `Launch Setting` task resulted in an error. Adding this task fixed the issue

**Comment by matthewnitschke** [comment]
@nicksnyder @felixfbecker Could someone take a look at this?

File: src/git.ts [hunk 3]
```diff
@@ -32,14 +32,21 @@ async function gitRemoteURL(repoDir: string, remoteName: string): Promise<string
...
+    // if there is no default remote configured, retrieve remote to use via 'git remote' command
+    if (remote === undefined) {
+        const currentRemotes = await gitRemotes(repoDir)
+        if (currentRemotes.length === 0) {
+            throw new Error('no configured git remotes')
+        }
+        if (currentRemotes.length > 1) {
+            log.appendLine(`using first git remote: ${currentRemotes[0]}`)
+        }
+        remote = currentRemotes[0]
```
Comment (nicksnyder) [inline src/git.ts]: reducing diff
```suggestion
    if (remote === undefined) {
        const remotes = await gitRemotes(repoDir)
        if (remotes.length === 0) {
            throw new Error('no configured git remotes')
        }
        if (remotes.length > 1) {
            log.appendLine(`using first git remote: ${remotes[0]}`)
        }
        remote = remotes[0]
```

**Comment by nicksnyder** [comment]
seems safe to merge, @sourcegraph/web can you own this?

**Comment by matthewnitschke** [comment]
Closing as I feel: https://github.com/sourcegraph/sourcegraph-vscode/pull/122 is a better solution to the problem

@nicksnyder Can someone take a look at that pr?

**Pull request closed without merging**